batchflow/models/validator/validator.py

```diff
@@ -0,0 +1,284 @@
+            used as first argument of `train`.
+         """
+        _ = kwargs
+        return path
+
+    def test_loader(self, path=None, **kwargs):
+        """ Test dataset loader.
+
+        Parameters
+        ----------
+        path : str or None
+            path to test dataset, defined in `test/dataset` section of config.
+
+        Return
+        ------
+        path : str or None
+            If function is not defined in child class, return `path`. It will be
+            used as the first argument of `train`.
+        """
+        _ = kwargs
+        return path
+
+    def load_model(self, path=None):
+        """ Loader for pretrained model.
+
+        Parameters
+        ----------
+        path : str
+
+        Return
+        ------
+        path : str
+            If function is not defined in child class, return `path` to model.
+            If `pretrained` is enabled, output of that function will be used
+            as the second argument of `inference`.
+        """
+        return path
+
+    def train(self, train_dataset, **kwargs):
+        """ Function that must contain the whole training process. If `pretrained`
+        is not enabled, output of that function will be used as the second argument
+        of `inference`. """
+        pass
+
+    def inference(self, test_dataset, train_output, **kwargs):
+        """ Function that must contain the whole inference process. The function must return
+        predictions and targets for metrics.
+        """
+        pass
+
+    def _compute_metrics(self):
+        """ Metrics computation. """
+        self.metric_values = {}
+        for _metric in self._metrics:
+            metric_class, params = key_value(_metric)
+            evaluate = params.pop('evaluate')
+            metrics = METRICS[metric_class](self.targets, self.predictions, **params)
+            self.metric_values[metric_class] = {}
+            for item in evaluate:
+                if isinstance(item, dict):
+                    metric_name = list(item.keys())[0]
+                    kwargs = list(item.values())[0]
+                else:
+                    metric_name = item
+                    kwargs = dict()
+                self.metric_values[metric_class][metric_name] = metrics.evaluate(metric_name, **kwargs)
+
+    def _compute_custom_metrics(self):
+        self.custom_metric_values = {}
+        for _metric in self._custom_metrics:
+            value = getattr(self, _metric)(self.targets, self.predictions)
+            self.custom_metric_values[_metric] = value
+
+    def run(self):
+        """ Run validator """
+        if 'train' in self.config:
+            self.train_dataset = self.train_loader(**self._train_ds)
+            self.from_train = self.train(self.train_dataset, **self.config['train'])
+        elif 'pretrained' in self.config:
+            self.from_train = self.load_model(**self._pretrained)
+
+        if 'test' in self.config:
```

Comment:
if not, it should throw a warning

Reply:
As I said before, classes can be without `train` or `inference` but we will use another one script to check whether class satisfy requirements or not.

Reply:
Why is validator needed when there is no validation?

Reply:
We can use `ModelAPI` just as wrapper for model. For example, it can be used to train and dump model.

Reply:
Why do we need a wrapper for a model when we have a model class?

Let's make it dead simple:
- Validator class is to validate a model
- one should define a child class with specific load, save, train, predict and custom metrics methods

LIghtningValidator can also be an example of a child Validator class.